naive_joint_space_planner.py

```
@@ -0,0 +1,469 @@
+        self.sq_dist_thresh = dist_thresh ** 2
+        self.eps = eps
+        self.dx, self.dy, self.dz = dx, dy, dz
+        # discretize continuous state space
+        self.dpos = np.array([dx, dy, dz])
+
+        # define action space
+        self.da = da_rad
+        self.num_joints = env.arm.num_joints
+        self.A = ActionSpace(num_DOF=self.num_joints, da_rad=self.da)
+        self.G = dict()
+        self.use_EE = False
+        self.guided_direction = True
+
+        # discretize continuous state/action space
+        self.xi_bounds = (
+            (self.xbounds - self.xbounds[0]) / self.dx).astype(int)
+        self.yi_bounds = (
+            (self.ybounds - self.ybounds[0]) / self.dy).astype(int)
+
+        self.visualize = visualize
+
+        # random samples of environmental parameters
+        self.num_rand_samples = num_rand_samples
+        self.sim_params_set = env.gen_random_env_param_set(
+            num=self.num_rand_samples)
+
+        # overall probability of a given simulation is product of probabilities
+        # of random env params
+        self.param_probs = [(p.bottle_fill_prob * p.bottle_fric_prob)
+                            for p in self.sim_params_set]
+
+        # safety threshold for prob(bottle fall) to deem invalid transitions
+        self.fall_proportion_thresh = 0.2
+
+        # method of simulating an action
+        self.sim_mode = sim_mode
+        if sim_mode == self.SINGLE:
+            self.sim_func = self.env.run_sim
+            assert(self.num_rand_samples == 1)
+        elif sim_mode == self.AVG:
+            self.sim_func = self.env.run_multiple_sims
+            self.process_multiple_sim_results = self.avg_results
+            # 1 is ok, can change, but to double-check that we're not using default value
+            assert(self.num_rand_samples > 1)
+        else:
+            print("Invalid sim mode specified: {}, defaulting to SINGLE".format(sim_mode))
```

Generally preferable to just bail out entirely if there's something as unexpected as this, rather than quietly covering for the user. Preferable to fail fast in unexpected conditions than let bad experiments continue.